core/modules/replica/StatusRequest.h

```diff
@@ -0,0 +1,453 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_STATUS_REQUEST_H
+#define LSST_QSERV_REPLICA_STATUS_REQUEST_H
+
+/// StatusRequest.h declares:
+///
+/// Common classes shared by all implementations:
+///
+///   class StatusRequestBase
+///   class StatusRequest
+///   class StatusRequestReplicate
+///   class StatusRequestDelete
+///   class StatusRequestFind
+///   class StatusRequestFindAll
+///
+/// Request implementations based on multiplexed connectors provided by
+/// base class RequestMessenger:
+///
+///   class StatusRequestBaseM
+///   class StatusRequestM
+///   class StatusRequestReplicateM
+///   class StatusRequestDeleteM
+///   class StatusRequestFindM
+///   class StatusRequestFindAllM
+///
+/// (see individual class documentation for more information)
+
+// System headers
+#include <functional>
+#include <future>
+#include <memory>
+#include <string>
+
+// Qserv headers
+#include "proto/replication.pb.h"
+#include "replica/Common.h"
+#include "replica/Messenger.h"
+#include "replica/ReplicaInfo.h"
+#include "replica/RequestMessenger.h"
+#include "replica/ProtocolBuffer.h"
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+// ========================================================================
+//   Customizations for specific request types require dedicated policies
+// ========================================================================
+
+struct StatusReplicationRequestPolicy {
+
+    static char const* requestTypeName() { return "REQUEST_STATUS:REPLICA_CREATE"; }
```

Comment:
nit: from the function name it's a little ambiguous if the function is requesting a type name or getting the type name of the request, I think it would be slightly less redundant and more specific if the name of the function was simply `name` or `typeName`

Reply:
Agree. Replaced with 'requestName' which is exactly what these strings are about.